File: frontend/src/types/zte.ts
export type ZTEConnection = {
  id: string
  name: string
  base_url: string
  olt_id: string
  is_active: boolean
  created_at: string
}

export type ZTESystemInfo = {
  oltId: string
  name: string
  host: string
  uptime: string
  cpuUsage: number
  memoryUsage: number
  isOnline: boolean
}

export type ZTEPONInfo = {
  board: number
  pon: number
  description: string
}

export type ZTEONUListItem = {
  oltId: string
  board: number
  pon: number
  onuId: number
  name: string
  serialNumber: string
  status: string
  statusCode: number
  rxPower: number
}

export type ZTEONUDetail = {
  oltId: string
  board: number
  pon: number
  onuId: number
  name: string
  serialNumber: string
  type: string
  status: string
  statusCode: number
  rxPower: number
  txPower: number
  lastOnline: string
  lastOffline: string
  offlineReason: string
  wanIp: string
}

File: internal/models/zte.go
package models

import "time"

type ZTEConnection struct {
	ID        string    `json:"id" db:"id"`
	Name      string    `json:"name" db:"name"`
	BaseURL   string    `json:"base_url" db:"base_url"`
	OltID     string    `json:"olt_id" db:"olt_id"`
	IsActive  bool      `json:"is_active" db:"is_active"`
	CreatedAt time.Time `json:"created_at" db:"created_at"`
}

type ZTEConnectionRequest struct {
	ID      string `json:"id"`
	Name    string `json:"name"`
	BaseURL string `json:"base_url"`
}

type ZTEConnectionUpdateRequest struct {
	Name    *string `json:"name,omitempty"`
	BaseURL *string `json:"base_url,omitempty"`
	OltID   *string `json:"olt_id,omitempty"`
}
